pkg/pushsync/pushsync.go

```diff
@@ -186,7 +213,18 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 		return fmt.Errorf("send receipt to peer %s: %w", peer.String(), err)
 	}
 
-	return ps.accounting.Debit(p.Address, ps.pricer.PriceForPeer(p.Address, chunk.Address()))
+	// Get price we charge for upstream peer read at headler
+	responseHeaders := stream.ResponseHeaders()
+	chunkPriceUpstream, err := headerutils.ReadPriceHeader(responseHeaders)
+
+	if err != nil {
+		// if not found in returned header, compute the price we charge for this chunk and
+		ps.logger.Warningf("No price in previously issued response headers")
```

Comment:
Error is omitted in log line. A chunk and peer addresses would be useful in this log line, also to add a prefix to indicate that tis line comes from the pushsync protocol.